include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,593 @@
+    }
+
+    // Exact search in current block.
+    else if (maxErrorsLeftInBlock == 0 && needleRightIt - needleLeftIt - 1 != s.blocklength[blockIndex])
+    {
+        _schemeSearchExact(delegate, iter, needle, needleLeftIt, needleRightIt, errors, s, blockIndex, TDir(), TDistanceTag());
+    }
+    // Approximate search in current block.
+    else // if (s.blocklength[blockIndex] - (needleRightIt - needleLeftIt - (needleLeftIt != needleRightIt)) >= minErrorsLeftInBlock)
+    {
+        // Insertion
+        if (std::is_same<TDistanceTag, EditDistance>::value)
+        {
+            bool const goToRight = std::is_same<TDir, Rev>::value;
+            int32_t const needleLeftIt2 = needleLeftIt - !goToRight;
+            uint32_t const needleRightIt2 = needleRightIt + goToRight;
+
+            if (needleRightIt - needleLeftIt == s.blocklength[blockIndex])
+            {
+                // leave the possibility for one or multiple deletions! therefore, don't change direction, etc!
+                _schemeSearchDeletion(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + 1, s, blockIndex, TDir());
+            }
+            else
+            {
+                _schemeSearch(delegate, iter, needle, needleLeftIt2, needleRightIt2, errors + 1, s, blockIndex, TDir(), TDistanceTag());
+            }
+        }
+        _schemeSearchChildren(delegate, iter, needle, needleLeftIt, needleRightIt, errors, s, blockIndex, minErrorsLeftInBlock, TDir(), TDistanceTag());
+    }
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks, typename TDistanceTag>
+inline void _schemeSearch(TDelegate & delegate,
+                          Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > it,
+                          TText2 const & needle,
+                          Search<nbrBlocks> const & s,
+                          TDistanceTag /**/)
+{
+    _schemeSearch(delegate, it, needle, s.startPos, s.startPos + 1, 0, s, 0, Rev(), TDistanceTag());
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks, size_t N, typename TDistanceTag>
+inline void _schemeSearches(TDelegate & delegate,
+                          Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > it,
+                          TText2 const & needle,
+                          std::array<Search<nbrBlocks>, N> const & ss,
+                          TDistanceTag /**/)
+{
+    for (auto & s : ss)
+        _schemeSearch(delegate, it, needle, s, TDistanceTag());
+}
+
+// ----------------------------------------------------------------------------
+// Function _find()
+// ----------------------------------------------------------------------------
+
+template <size_t minErrors, size_t maxErrors, typename TText, typename TIndexSpec, typename TPattern, typename TDelegate, typename TDistanceTag>
```

Comment:
Please keep the order of template arguments in sync with the order of function arguments.